src/bounds.rs

```diff
@@ -361,22 +361,25 @@ mod tests {
         assert_eq!(exp, Bounds::try_from([1.0, 2.0, 3.0, 4.0].as_slice())?);
         assert_eq!(exp, Bounds::try_from(vec![1.0, 2.0, 3.0, 4.0])?);
         let val = vec![1.0, 2.0, 3.0, 4.0];
-        assert_eq!(exp, Bounds::try_from((&val).as_slice())?);
+        let borrowed = &val;
+        assert_eq!(exp, Bounds::try_from(borrowed.as_slice())?);
```

Comment:
```suggestion
        assert_eq!(exp, Bounds::try_from(val.as_slice())?);
```

Reply:
I'm confused why clippy recommended that.

Reply:
I might be over-careful here to try all possible usages. I used to have `(&val).as_slice()` there, which is what clippy was complaining about.